### DISCUSSION TITLE
Possible to query the same object from different queries and make them same type?

### DISCUSSION THREAD

**Discussion by pjwalstrom** [graphql-java/graphql-java#2200] [Q&A] [answered]
Hello,
we are currently using [graphql-kotlin](https://github.com/ExpediaGroup/graphql-kotlin/), but we face some limitations which we are wondering might be possible to overcome with graphql-java. 

we have an external schema on which we have no possibility to make changes. The schema may look like this
```
type Query {
    foo: Bar
}
type Bar {
    qux: String
}
```
Then we have two different queries, let's say MyFirstQuery.graphql and MySecondQuery.graphql, that look the same
```
query {
    foo {
        qux
    }
}
```
With graphql-kotlin, we end up with generated Kotlin-classes, `MyFirstQuery` and `MySecondQuery`, that both encapsulate the `data class Bar`. Which essentially is the same class, but the type system makes them different.

Is there a way to make `MyFirstQuery.Bar.class == MySecondQuery.Bar.class` with graphql-java?

I asked the question over at the graphql-kotlin repo here: https://github.com/ExpediaGroup/graphql-kotlin/discussions/1039

**Comment by andimarek** [comment, marked as answer]
This seem like a graphql-kotlin question: GraphQL Java doesn't do anything with classes and we also don't map them to anything. 

GraphQL Java has DataFetcher per field and what you return per DataFetcher needs to map the schema at execution time.

**Reply by pjwalstrom** [reply]
great, thanks, I will look into this!

**Reply by pjwalstrom** [reply]
no way to generate classes based on the schema?

**Reply by bbakerman** [reply]
> no way to generate classes based on the schema?

Again graphql-java is the base library that enables a bunch of other libraries.

We do not do higher level application concerns like generate classes from schema - but other libs do

**Reply by andimarek** [reply]
If you are looking for code generated by schema, have a look at this recently released project by netflix: https://github.com/Netflix/dgs-framework/
